torchtitan/parallelisms/pipeline_llama.py

```diff
@@ -141,6 +146,30 @@ def _build_stage(stage_idx, start_layer, stop_layer, is_first=False, is_last=Fal
         )
         return stage, model
 
+    # if split points are not specified, we split the model into equal chunks based on
+    # the number of pipeline stages.
+    if len(splits) == 0:
+        # we assume num_stages per rank based on the schedule time
+        schedule_class = get_schedule_class(
+            job_config.experimental.pipeline_parallel_schedule
+        )
+        if issubclass(schedule_class, PipelineScheduleSingle):
+            num_stages_per_rank = 1
+        elif issubclass(schedule_class, PipelineScheduleMulti):
+            num_stages_per_rank = 2
```

Comment:
Curious in principle, can # stages / rank be higher today? It sounds to me that setting this higher would further reduce bubble but requires more layers in the model.
Maybe it helps if we add a comment here, saying this is just a default choice and is not a restriction for manual split.

Reply:
Yep you can increase the # stages for multistage schedules. It would result in more communication. I will add a comment.